### ISSUE TITLE
Model_Soft - reference key is deleted or not

### ISSUE THREAD

**Issue by n-kaito** [fuel/orm#278]
I'm using `Model_Soft`
When I use the delete function, the results are different when I refer to the releted model or not 

Please see [my sample at Gist](https://gist.github.com/n-kaito/15617119ec83e65a524b)

version of fuel/orm is `5105b984d4656f81d06296817ac0d19be573abf7`

**Comment by WanWizard** [comment]
@stevewest was this addressed?

**Comment by emlynwest** [comment]
No, not had the time recently. Going to try and make time over the next couple of weeks.

**Comment by emlynwest** [comment]
Does this only happen with soft models or does it happen with regular models as well? The FK updating is done by the base model at the moment, not Model_Soft.
